index.bs

```diff
@@ -2740,27 +4781,107 @@ partial interface MLGraphBuilder {
+    : <dfn>mode</dfn>
+    ::
+        An {{MLInterpolationMode}} [=string=].
+        Specifies the interpolation algorithm used to fill the output tensor values.
+        The default value is `"nearest-neighbor"`, standing for *Nearest Neighbor* interpolation.
+
+    : <dfn>scales</dfn>
+    ::
+        A sequence of {{float}} of length 2.
+        Specifies the scaling factor in each spatial dimensions of the input: [scale_height, scale_width].
+        The default value is [1.0, 1.0].
+
+    : <dfn>sizes</dfn>
+    ::
+        A sequence of {{unsigned long}} of length 2.
+        Specifies the target sizes for each spatial dimensions of the input: [size_height, size_width]. When the target sizes are specified, the {{MLResample2dOptions/scales}} argument is ignored, since the scaling factor values are derived from the target sizes of each spatial dimension of the input.
+
+    : <dfn>axes</dfn>
+    ::
+        A sequence of {{unsigned long}} of length 2.
+        Specifies the two consecutive dimensions of the input tensor to which the interpolation algorithm applies. The valid values in the sequence are [0, 1], [1, 2] or [2, 3].
+        The default value is [2, 3].
+</dl>
+
+<details open>
+  <summary>
+    To <dfn for="MLGraphBuilder">check resample options</dfn> given |options|, run the following steps:
+  </summary>
+  <div algorithm=check-resample-options class=algorithm-steps>
+    1. If |options| is `undefined`, let |options| be a new {{MLResample2dOptions}} object.
+    1. If |options|.{{MLResample2dOptions/mode}} [=map/exists=]:
+        1. If its value is not one of `"nearest-neighbor"` or `"linear"`, return `null`.
+    1. Otherwise, set |options|.{{MLResample2dOptions/mode}} to `"nearest-neighbor"`.
+    1. If |options|.{{MLResample2dOptions/scales}} [=map/exists=]:
+        1. If its size is not `2`, or if any of its values is not greater than `0`, return `null`.
+    1. Otherwise, set |options|.{{MLResample2dOptions/scales}} to `[1.0, 1.0]`.
+    1. If |options|.{{MLResample2dOptions/sizes}} [=map/exists=]: if its size is not `2`, or if any of its values is not greater than `0`, return `null`.
+    1. If |options|.{{MLResample2dOptions/axes}} [=map/exists=]:
+        1. If its value is not one of `[0, 1], [1, 2], [2, 3]`, return `null`.
+    1. Otherwise, set |options|.{{MLResample2dOptions/axes}} to `[2, 3]`.
+    1. Return |options|.
+  </div>
+</details>
+
+<details open>
+  <summary>
+    To <dfn for="MLGraphBuilder">resample output sizes</dfn> given |input| and |options|, run the following steps:
+  </summary>
+  <div algorithm=resample-output-sizes class=algorithm-steps>
+    1. Let |desc| be an {{MLOperandDescriptor}} initialized to |input|.{{MLOperand/[[descriptor]]}}.
+    1. If |options|.{{MLResample2dOptions/sizes}} [=map/exists=], then set |desc|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} to |options|.{{MLResample2dOptions/sizes}} and return |desc|.
```

Comment:
The size of |options|.{{MLResample2dOptions/sizes}} must be 2 and the value of |options|.{{MLResample2dOptions/sizes}} indicated the output size of the specified dimensions given by {{MLResample2dOptions/axes}}. For the remaining dimensions, the value of output size is equal to the input size.